core/src/test/java/brooklyn/location/geo/HostGeoLookupIntegrationTest.java

```diff
@@ -34,6 +35,7 @@
 
     public static final Logger log = LoggerFactory.getLogger(HostGeoLookupIntegrationTest.class);
     
+    // Needs fast network connectivity to figure out the external IP. If response not returned in 2s fails.
```

Comment:
Can't wait longer/differentiate the result here. It's designed to avoid long blocks on purpose and fail gracefully.
We are already testing all implementations separately so it's worth investigating if we can use stubs here to test the higher level logic.